.github/CONTRIBUTING.md

```diff
@@ -1,71 +1,167 @@
-# Contributing to `terraform-provider-tls`
+# Contributing to the TLS Provider
 
-**First:** if you're unsure or afraid of _anything_, just ask or submit the issue describing the problem you're aiming to solve.
+Thank you for investing your time in contributing to our project: please ensure you are familiar
+with [HashiCorp Code of Conduct](https://github.com/hashicorp/.github/blob/master/CODE_OF_CONDUCT.md).
 
-The `tls` provider is a HashiCorp *standard library* provider, which means we consider it part of the core Terraform experience. Our priority is stability and correctness, and any bug fix or feature has to be considered in the context of this provider's many users and the wider Terraform ecosystem.
-This is great as your contribution can have a big positive impact, but we have to assess potential negative impact too.
+This provider is an HashiCorp **utility provider**, which means any bug fix and feature
+has to be considered in the context of the thousands/millions of configurations in which this provider is used.
+This is great as your contribution can have a big positive impact, but we have to assess potential negative impact too
+(e.g. breaking existing configurations).
 
-To provide some safety to the Terraform ecosystem, we strictly follow [semantic versioning](https://semver.org/) and any changes that could be considered as breaking will only be released as part of a major release.
+To provide some safety to the wider provider ecosystem, we strictly follow
+[semantic versioning](https://semver.org/) and HashiCorp own
+[versioning specification](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#versioning-specification).
+Any changes that could be considered as breaking will only be released as part of major release.
 
-## Table of Contents
+## Identifying the correct scope
 
- - [I just have a question](#i-just-have-a-question)
- - [I want to report a vulnerability](#i-want-to-report-a-vulnerability)
- - [New Issue](#new-issue)
- - [New Pull Request](#new-pull-request)
+While Terraform acts as a single program from the user's perspective
+it is made up of a few parts, each of which have different role and repository.
 
-## I just have a question
+This section describes the scope of notable repositories, which may help you
+ensure you're in the right place when reporting bugs and feature requests,
+or submitting a patch.
 
-> **Note:** We use GitHub for tracking bugs and feature requests only.
+* `hashicorp/terraform` - Terraform **Core** which implements all the low-level functionality which
```

Comment:
ow about:

Terraform **Core** implements all the low-level functionality that isn't domain specific (covered by providers).